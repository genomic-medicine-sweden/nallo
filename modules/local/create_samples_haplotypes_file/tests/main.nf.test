nextflow_process {

    name "Test Process CREATE_SAMPLES_HAPLOTYPES_FILE"
    script "../main.nf"
    process "CREATE_SAMPLES_HAPLOTYPES_FILE"

    tag "modules"
    tag "modules_local"
    tag "create_samples_haplotypes_file"

    config "./nextflow.config"

    test("create_samples_haplotypes_file") {

        when {
            process {
                """
                input[0] = Channel.of('test_vcf_file_sample')
                    .collectFile(name: 'test.txt')
                    .map { file -> [ [ id: 'test' ], file ] }
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("create_samples_haplotypes_file -stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = Channel.of('test_vcf_file_sample')
                    .collectFile(name: 'test.txt')
                    .map { file -> [ [ id: 'test' ], file ] }
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }
}
