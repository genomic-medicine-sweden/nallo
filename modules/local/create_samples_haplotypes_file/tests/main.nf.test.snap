{
    "create_samples_haplotypes_file": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "samples_file.txt:md5,0c38c6a80221e7b24150e38de89f0e14"
                    ]
                ],
                "1": [
                    "versions.yml:md5,6dc2df11aeedff4dafd2c4f2d646080e"
                ],
                "samples": [
                    [
                        {
                            "id": "test"
                        },
                        "samples_file.txt:md5,0c38c6a80221e7b24150e38de89f0e14"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,6dc2df11aeedff4dafd2c4f2d646080e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-07-04T10:41:07.977475767"
    },
    "create_samples_haplotypes_file -stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "samples_file.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,6dc2df11aeedff4dafd2c4f2d646080e"
                ],
                "samples": [
                    [
                        {
                            "id": "test"
                        },
                        "samples_file.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,6dc2df11aeedff4dafd2c4f2d646080e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-07-04T10:41:13.114623353"
    }
}